Makefile

```diff
@@ -1,35 +1,41 @@
+# Currently always builds for amd64
+#PLATFORM_ARCH=$(shell uname -m | tr '[:upper:]' '[:lower:]')
+PLATFORM_ARCH=amd64
+PLATFORM_NAME=$(shell uname -s | tr '[:upper:]' '[:lower:]')
+VERSION_TAG=$(shell cat .go-version)
+
 .PHONY: build
 build:
-	@go build -o terraform-provider-lightstep_v$(shell cat .go-version)
+	GOARCH=$(PLATFORM_ARCH) go build -o terraform-provider-lightstep_v$(VERSION_TAG)
```

Comment:
Historically it looks like we've always hard-coded the local build to `amd64`. The folders and filenames have been hard-coded to that.  However, it was not setting a flag on the go compiler so binaries were being produced for "amd64" even though they were compiled as "arm64" on my M1.  This fixes that.

The rest was a bit of clean-up to use named variables instead of repeated shell commands.

Lastly, I removed some of the `@` prefixes to not silence the Make commands. Maybe that's subjective, but I find it frustrating when a make command fails to have no output of what it was running!